src/crypto/mod.rs

```diff
@@ -81,17 +80,33 @@ impl EcKeyComponents {
     }
 }
 
-pub trait Crypto: Sized {
-    type RemotePublicKey: RemotePublicKey;
```

Comment:
Removed to associated types to make `Cryptographer` object-safe.

Reply:
Note: which is also why we end up with https://github.com/mozilla/rust-ece/pull/36#discussion_r313172900.